File: python/easy/detect_capital.py
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        template = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        cap_cnt = 0
        pos = 0
        idx = 0
        for idx in range(0, len(word)):
            if word[idx] in template:
                cap_cnt += 1
                pos = idx
        if cap_cnt == len(word):
            return True
        elif cap_cnt == 1 and pos == 0:
            return True
        elif cap_cnt == 0:
            return True
        else:
            return False

File: c++/easy/detect_capital.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        string temp = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        int pos = 0;
        int capital_cnt = 0;
        for(int i = 0; i < word.size(); ++i)
        {
            int is_in =temp.find(word[i]);
            if(is_in != -1)
            {
                ++capital_cnt;
                pos = i;
            }
        }
        if(capital_cnt == 0)
            return true;
        else if(capital_cnt == word.size())
            return true;
        else if(capital_cnt == 1 && pos == 0)
            return true;
        else
            return false;
    }
};

Conclustion:
create a template string (from 'A' to 'Z') and to check whether or not each charactor in the template string.
time complexity O(n)
space complexity O(1)
